feat!: dbtp 928 add cdn endpoint module (#141)
uktrade/terraform-platform-modules: README.md

example `extensions.yml` config.

```yaml
my-application-alb:
  type: alb
  environments:
    dev: 
      additional_address_list:
        - internal.my-web-service-2
```

## CDN

This module will create the CloudFront (CDN) endpoints for the application if enabled.

> Additional domains (`cdn_domains_list`) are the domain names that will be configured in CloudFront. In the map the key is the fully qualified domain name and the value is the application's base domain (the application's Route 53 zone).  

Reviewer: It's confusing to us "Additional domains" here.

Also, the instruction on the value appears to be wrong based on the information under the example. Can the be lifted up here and removed from below?
```suggestion
`cdn_domains_list` is a map of the domain names that will be configured in CloudFront.
* the key is the fully qualified domain name
* the value is an array containing the internal prefix and the base domain (the application's Route 53 zone).  
```

example `extensions.yml` config.

```yaml
my-application-alb:
  type: alb
  environments:
    dev: 
      cdn_domains_list:
        - dev.my-application.uktrade.digital: [ "internal", "my-application.uktrade.digital" ] 
        - dev.my-web-service-2.my-application.uktrade.digital: [ "internal.my-web-service-2", "my-application.uktrade.digital" ]
      additional_address_list:
        - internal.my-web-service-2
      enable_cdn_record: false
      enable_logging: true
    prod:

>       cdn_domains_list: {my-application.great.gov.uk: "great.gov.uk"} 

Reviewer 2: Is it a problem that this isn't a list?

Author: Its a list of key: values eg.
{my-application.great.gov.uk: "great.gov.uk", my-secondapp.great.gov.uk: "great.gov.uk"}

Author: @gabelton is that ok, or am I missunderstanding your comment?

Reviewer 2: but it's technically a dict / map? Should it not be in the same format as `cdn_domains_list` under dev above ?

Author: oh my, you are correct.  I missed this, updaing.

Author: updated

```

> Each item in the `cdn_domain_list` must include:
> - Key: The endpoint name <myapp.mysite.com>
> - Values: application internal prefix and base domain <"internal", "mysite.com">

Reviewer: ```suggestion
```

### Optional settings:

> To create a R53 record pointing to the CloudFront endpoint set this to true.  If not set, in non production this is set to true by default and set to false in production.

Reviewer 2: ```suggestion
To create a R53 record pointing to the CloudFront endpoint, set this to true.  If not set, in non production this is set to true by default and set to false in production.
```

- enable_cdn_record: true

> To turn on CloudFront logging to a S3 bucket set this to true.

Reviewer 2: ```suggestion
To turn on CloudFront logging to a S3 bucket, set this to true.
```

> - enable_logging: true

Reviewer: I would be inclined move this above the example, I just deleted the two questions I already asked there which this answers.

Author: moved
